src/Management/src/Endpoint/Services/ServiceContextDescriptor.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Steeltoe.Management.Endpoint.Services;
+
+public class ServiceContextDescriptor:Dictionary<string, ApplicationContext>
```

Comment:
It would be nice if the shape matches what the serializer needs. That would avoid reflection calls each time the endpoint is hit. The data structure can be populated from ConfigureOptions, assuming the collection is immutable once the application has started.

Reply:
The spring side has a cache time to live that would be worth considering , but where are you seeing reflection used?

Reply:
fixed with this commit: https://github.com/SteeltoeOSS/Steeltoe/pull/1129/commits/c1d49201f13a856d296a5a5ab8155cf7218d929e